### DISCUSSION TITLE
230VAC 5 optocoupler inputs with 5 relays outputs

### DISCUSSION THREAD

**Discussion by VerreyckenGerd** [arendst/Tasmota#18278] [Show and tell]
I made these boards to integrate my teleruptor based home light system into my node-red automation environment.

![image](https://user-images.githubusercontent.com/69856911/227931573-67e36c52-aeed-4bd2-8dbb-6f194da3609f.png)

The 5 optocoupler inputs read the presence of mains voltage on the outputs of my teleruptors. The 5 relais are in parralel with my lighting pushbuttons and will switch the teleruptors position. This way if my automation system fails I can still operate the lights as they where originally. The wemos D1 mini's contain Tasmota and us MQTT to talk to my Node-red that runs on a dedicated Raspberry PI.

**Comment by VerreyckenGerd** [comment]
Here is the schematic if anyone wants to build something similar
![image](https://user-images.githubusercontent.com/69856911/227933274-d943cb35-961c-4a55-9961-0b36b83d0df9.png)

**Reply by barbudor** [reply]
Nice
Why the 47nF is serie with the input?

I see the LM7833 creates a 3v3 from 5V but isn't already a regulator on the Wemos? That sounds like a possible conflict between those 2.

**Reply by VerreyckenGerd** [reply]
Indeed the choise has to be made to use the onboard wemos psu or an external one by leaving the 7833 out or leave the pin 3,3V from the wemos board. The 47nF makes an AC coupling for the LED inside the optocoupler, lets only a certain charge trough so the series resistor has less energy to absorb.

**Comment by VerreyckenGerd** [comment]
![20220606_200558](https://user-images.githubusercontent.com/69856911/227934744-674da2c4-f55c-4ef6-8c5f-bc7f06066464.jpg)
This is how it looks in real life

**Reply by barbudor** [reply]
Good old pass-through parts. No SMDs 👍

**Comment by Tolo81** [comment]
Love This type of PCB

**Comment by Noschvie** [comment]
Hello
how can this PCB be "reused" to support 24-30V AC instead of 220V as input for the optocouplers ?
Background: status of irrigation valves to MQTT
Thanks!

**Reply by VerreyckenGerd** [reply]
By bridging the 47nF caps and changing the AC/DC convertor to a type suitable for taking 24VAC instead of 240VAC..

**Comment by salvatore640** [comment]
Hi, sorry for bothering you.  Is it possible to know how you integrate it with Home Assistant?  Thank you

**Reply by VerreyckenGerd** [reply]
Well, I didn't I have no Home Assistant running. I built my home automation entirely around Node-Red.
